src/video_core/buffer_cache/buffer_cache.cpp

```diff
@@ -235,25 +235,33 @@ bool BufferCache::BindVertexBuffers(
 }
 
 u32 BufferCache::BindIndexBuffer(bool& is_indexed, u32 index_offset) {
-    // Emulate QuadList primitive type with CPU made index buffer.
+    // Emulate QuadList and Polygon primitive types with CPU made index buffer.
     const auto& regs = liverpool->regs;
-    if (regs.primitive_type == AmdGpu::PrimitiveType::QuadList && !is_indexed) {
-        is_indexed = true;
-
+    if (!is_indexed) {
         // Emit indices.
         const u32 index_size = 3 * regs.num_indices;
         const auto [data, offset] = stream_buffer.Map(index_size);
-        Vulkan::LiverpoolToVK::EmitQuadToTriangleListIndices(data, regs.num_indices);
+
+        switch (regs.primitive_type) {
+        case AmdGpu::PrimitiveType::QuadList:
+            Vulkan::LiverpoolToVK::EmitQuadToTriangleListIndices(data, regs.num_indices);
+            break;
+        case AmdGpu::PrimitiveType::Polygon:
+            Vulkan::LiverpoolToVK::EmitPolygonToTriangleListIndices(data, regs.num_indices);
+            break;
+        default:
+            return regs.num_indices;

```

Comment:
I think the way this is laid out there's a bit of waste doing the stream buffer map, would be nice if we could terminate earlier if we don't need to generate a buffer.